### PR TITLE
Add simulated mode. Closes #167

### LINKED ISSUE #167: Simulated mode

### PR DESCRIPTION
**Pull request by hueniverse**

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -147,6 +147,24 @@ the `request.auth.isAuthenticated` must be checked to test if authentication fai
...
 
+#### Simulated authentication
+
+Testing applications using third-party login can be challenging given the lack of user interaction to perform the third-party login
+flow as well as the multiple steps required. To assist in testing such application without having to modify the application with custom
+code, **Bell** provides an override method `Bell.simulate()` which puts the module into simulation mode and any strategies created while
+it is in this mode will return the simulated credentials.
+
+The `Bell.simulate(credetialsFunc)` takes a single argument:
+- `credetialsFunc` - a function called for each incoming request to the protected resource with the signature `function(request, next)` where:
```
Comment (ldesplat) [inline API.md]: credentialsFunc

**Comment by hueniverse** [comment]
Fixed, but I am not sure this makes sense anymore... :-)

I forgot I can just pass the credentials I want to `server.inject()`. Feel free to merge this but if you are testing your server with `server.inject()` you don't actually need it. If you need to start your server to test it with an actual connection, this might be useful.

**Comment by ldesplat** [comment]
I'll merge it since it sounds like it's useful. I won't make a release just yet unless you change your mind and need it right now :)

**Comment by ldesplat** [comment]
Thank you! Can't believe I forgot!

**Pull request merged**